src/controllers/product.controller.js

```diff
@@ -108,6 +110,41 @@ export const getSpecificProduct = async (req, res) => {
     return res.status(500).json({
       error: err.message,
       message: 'Error occured while retrieving a product',
+      
+    });
+  }
+};
+export const markAvailableProduct = async (req, res) => {
+  try {
+    const {isAvailable} = req.body;
+    await ProductService.updateProduct( isAvailable, req.params.productId);
+   return res.status(200).json({ message: 'product is available now' });
+   
+  } catch (error) {
+    return res.status(500).json({
+      error: error.message,
+      message: 'Failed to mark if product is available or not',
     });
   }
 };
+export const updateProduct = async (req, res) => {
+  try{
+     await ProductService.updateProduct(req.body, req.params.productId);
```

Comment:
You don't have to use req.params, when you have a middleware that returns the product